crates/core/tedge_agent/src/file_transfer_server/actor.rs

```diff
@@ -0,0 +1,146 @@
+use crate::file_transfer_server::http_rest::http_file_transfer_server;
+use crate::file_transfer_server::http_rest::HttpConfig;
+use async_trait::async_trait;
+use std::convert::Infallible;
+use tedge_actors::Actor;
+use tedge_actors::Builder;
+use tedge_actors::DynSender;
+use tedge_actors::NoMessage;
+use tedge_actors::RuntimeError;
+use tedge_actors::RuntimeRequest;
+use tedge_actors::RuntimeRequestSink;
+use tedge_actors::SimpleMessageBoxBuilder;
+use tracing::log::error;
+
+pub struct FileTransferServerActor {
+    config: HttpConfig,
+}
```

Comment:
There is nothing wrong to have an actor with no message boxes, no receivers, no senders ... except that this actor can no more interact with others including the runtime. One need a least a way to receive `Shutdown` request.

```suggestion
pub struct FileTransferServerActor {
    config: HttpConfig,
    signal_receiver: mpsc::Receiver<RuntimeRequest>,
}
```